### PR TITLE
Remove host name if sharing with referencing xref

### PR DESCRIPTION
**Pull request by live1206**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/117912

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5071)

### CODE REVIEW

File: src/docfx/build/xref/XrefResolver.cs [hunk 1]
```diff
@@ -57,8 +58,8 @@ public XrefResolver(Context context, Docset docset, RestoreFileMap restoreFileMa
             {
                 queries["view"] = moniker;
             }
-            var resolvedHref = UrlUtility.MergeUrl(
-                xrefSpec.Href,
+            resolvedHref = UrlUtility.MergeUrl(
+                resolvedHref,
```
Comment (mingwli) [inline src/docfx/build/xref/XrefResolver.cs]:
```suggestion
                RemoveSharingHost(xrefSpec.Href, referencingFile.Docset.HostName),
```

File: src/docfx/build/xref/XrefResolver.cs [hunk 2]
```diff
@@ -130,6 +131,16 @@ public XrefMapModel ToXrefMapModel()
             return model;
         }
 
+        private string RemoveSharingHost(string url, string hostName)
+        {
+            if (url.StartsWith(hostName, StringComparison.OrdinalIgnoreCase))
```
Comment (yufeih) [inline src/docfx/build/xref/XrefResolver.cs]:
```suggestion
            if (url.StartsWith(hostName + '/', StringComparison.OrdinalIgnoreCase))
```

Comment (live1206) [reply src/docfx/build/xref/XrefResolver.cs]: why do we need appending `/` with hostName here?

Comment (yufeih) [reply src/docfx/build/xref/XrefResolver.cs]: `https://a.com/b` starts with `https://a.co`, but you shouldn't remove in this case 😁

Comment (live1206) [reply src/docfx/build/xref/XrefResolver.cs]: Thanks for the explanation, fair enough

**Pull request merged**